calcrule_validations/calculation_rule.py

```diff
@@ -49,40 +49,48 @@ def ready(cls):
 
     @classmethod
     def run_calculation_rules(
-        cls, sender, validation_class,
+        cls, sender, instance,
         user, context, **kwargs
     ):
         field_name = kwargs.pop('field_name')
         field_value = kwargs.pop('field_value')
         return cls.calculate_if_active_for_object(
-            validation_class,
+            instance,
             field_name=field_name,
             field_value=field_value,
             **kwargs
         )
 
     @classmethod
     def calculate_if_active_for_object(
-        cls, validation_class, calculation_uuid, **kwargs
+        cls, instance, calculation_uuid=None, **kwargs
     ):
+        if not calculation_uuid:
+            return False
         field_name = kwargs.pop('field_name')
         field_value = kwargs.pop('field_value')
-        if cls.active_for_object(validation_class, calculation_uuid):
-            return cls.calculate(validation_class, field_name, field_value, **kwargs)
+        if cls.active_for_object(instance, calculation_uuid):
+            return cls.calculate(instance, field_name, field_value, **kwargs)
 
     @classmethod
-    def active_for_object(cls, validation_class, calculation_uuid):
-        return cls.check_calculation(validation_class, calculation_uuid)
+    def active_for_object(cls, instance, calculation_uuid=None):
```

Comment:
This is still not in line with what AbsStrategy requires: https://github.com/openimis/openimis-be-core_py/blob/develop/core/abs_calculation_rule.py#L102

Reply:
indeed, it will not happen until someone will use this calculation elsewhere. As I explained to Damian we should drop the approach using the abstract and focus on Validation only (this is an helper to access BaseValidation) but this has impact on social protection so I did not want to do this kind on change.